base/src/jvmMain/kotlin/text/DateFormatter.kt

```diff
@@ -126,9 +127,13 @@ actual class DateFormatter private constructor(private val format: SimpleDateFor
 
     actual fun format(date: Date): String = format.format(date.calendar.time)
     actual fun parse(string: String): Date? {
-        return format.parse(string)?.let { date ->
-            val calendar = format.calendar.clone() as Calendar
-            Date(calendar.apply { time = date })
+        return try {
+            format.parse(string)?.let { date ->
+                val calendar = format.calendar.clone() as Calendar
+                Date(calendar.apply { time = date })
+            }
+        } catch (e: ParseException) {
+            null
```

Comment:
Would be good to add a little test of parsing some incorrect string to see all platforms have the same result (`null`)